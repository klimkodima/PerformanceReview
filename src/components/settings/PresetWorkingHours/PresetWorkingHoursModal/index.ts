export { default as PresetWorkingHoursModal } from './PresetWorkingHoursModal';
