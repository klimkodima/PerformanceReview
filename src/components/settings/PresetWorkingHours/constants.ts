import { t } from 'i18next';

import { ConstantsTextType } from '../types';

export const HEADER_CELLS: string[] = [
  t('preset_working_hours_table.header_cell_label'),
  t('preset_working_hours_table.header_cell_hours')
];

export const ERROR_TEXTS: ConstantsTextType = {
  emptyValueError: t('preset_working_hours_table.empty_value_error'),
  uniqueLabelError: t('preset_working_hours_table.unique_label_error'),
  hoursPatternError: t('preset_working_hours_table.hours_pattern_error'),
  hoursValueError: t('preset_working_hours_table.hours_value_error')
};

export const MODAL_ERROR_TEXTS: ConstantsTextType = {
  exceedsValueNotification: t(
    'preset_working_hours_table.modal_notification_exceeds'
  ),
  lessValueNotification: t(
    'preset_working_hours_table.modal_notification_less'
  ),
  emptyValueNotification: t(
    'preset_working_hours_table.modal_notification_empty_table'
  )
};

export const OPTIONS_TEXTS: ConstantsTextType = {
  optionValue1: t('preset_working_hours_table.option_value_1'),
  optionValue2: t('preset_working_hours_table.option_value_2'),
  optionValue3: t('preset_working_hours_table.option_value_3')
};

export const HOURS_PER_DAY = 8;
