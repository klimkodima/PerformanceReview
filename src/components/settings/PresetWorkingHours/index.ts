export { default as PresetWorkingHours } from './PresetWorkingHours';
export { default as PresetWorkingHoursEdit } from './PresetWorkingHoursEdit';
