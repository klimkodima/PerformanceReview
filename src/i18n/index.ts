import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import common from './locales.en.json';

const resources = {
  en: {
    translation: common
  }
};

const setI18nextConfig = async () =>
  await i18n.use(LanguageDetector).use(initReactI18next).init({
    resources,
    fallbackLng: 'en'
  });

void setI18nextConfig();

export default i18n;
