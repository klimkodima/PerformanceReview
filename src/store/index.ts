import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

const sagaMiddleware = createSagaMiddleware();

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['dateFilter', 'tasksCoefficient']
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    }).concat(sagaMiddleware)
});

sagaMiddleware.run(rootSaga);

export type RootStateType = ReturnType<typeof rootReducer>;
export type StoreType = typeof store;

export const persistor = persistStore(store);
export default store;
