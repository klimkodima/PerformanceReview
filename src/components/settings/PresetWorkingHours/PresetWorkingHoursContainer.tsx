import { memo, ReactElement, useState, useEffect } from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { nanoid } from 'nanoid';
import { isEmpty } from 'lodash';
import {
  TableContainer,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Checkbox,
  FormControl,
  Select
} from '@mui/material';

import { ErrorTooltip } from 'src/components/common';
import {
  fetchPresetWorkingHours,
  selectPresetWorkingHours,
  postPresetWorkingHours,
  putPresetWorkingHours,
  deletePresetWorkingHours
} from 'src/store/PresetWorkingHours';
import {
  HEADER_CELLS,
  ERROR_TEXTS,
  MODAL_ERROR_TEXTS,
  OPTIONS_TEXTS,
  HOURS_PER_DAY
} from './constants';
import { PresetWorkingHoursModal } from './PresetWorkingHoursModal';
import { SettingsTableBtnsBlock } from '../common';
import { handleCheckboxClick, onChangeInput } from '../common/helpers';

import { PresetWorkingHoursDataType } from 'src/store/PresetWorkingHours/types';

type UsersContainerPropsType = {
  isEditable: boolean;
};

const UsersContainer = memo(
  ({ isEditable }: UsersContainerPropsType): ReactElement => {
    const rowsApi = useSelector(selectPresetWorkingHours);
    const dispatch = useDispatch();

    useEffect(() => {
      dispatch(fetchPresetWorkingHours());
    }, []);

    const [isEdit, setIsEdit] = useState<boolean>(false);
    const [isValid, setIsValid] = useState(false);
    const copyRowsApi: PresetWorkingHoursDataType[] = JSON.parse(
      JSON.stringify(rowsApi)
    );

    const [rowsState, setRowsState] =
      useState<PresetWorkingHoursDataType[]>(copyRowsApi);

    useEffect(() => {
      setRowsState(copyRowsApi);
    }, [rowsApi]);

    const [selected, setSelected] = useState<string[]>([]);
    const rows = isEdit ? rowsState : copyRowsApi;
    const editClassName = isEdit ? 'settings-table__edit' : '';
    const {
      emptyValueError,
      uniqueLabelError,
      hoursPatternError,
      hoursValueError
    } = ERROR_TEXTS;
    const emptyRow = {
      id: nanoid(),
      label: {
        value: 'Audits',
        isError: false,
        errorText: ''
      },
      time: {
        value: '',
        isError: false,
        errorText: ''
      }
    };
    const { optionValue1, optionValue2, optionValue3 } = OPTIONS_TEXTS;

    const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
    const {
      exceedsValueNotification,
      lessValueNotification,
      emptyValueNotification
    } = MODAL_ERROR_TEXTS;
    const [isLess, setIsLess] = useState<boolean>(false);
    const [isEmptyTable, setIsEmptyTable] = useState<boolean>(false);
    let modalNotification = isLess
      ? lessValueNotification
      : exceedsValueNotification;
    if (isEmptyTable) modalNotification = emptyValueNotification;

    const addNewRow = (): void => {
      setRowsState([...rowsState, emptyRow]);
    };

    const checkValidation = (): void => {
      // !unique status
      const statusNames: string[] = [];
      rowsState.forEach((row: PresetWorkingHoursDataType) =>
        statusNames.push(row.label.value.trim().toLowerCase())
      );
      const uniqueStatusNames = new Set(statusNames);
      let notUniqueStatus: string[] = [];

      if (uniqueStatusNames.size < statusNames.length) {
        const countStatusValues = statusNames.reduce((acc, item) => {
          acc[item] = acc[item] ? Number(acc[item]) + 1 : 1;
          return acc;
        }, {});

        notUniqueStatus = Object.keys(countStatusValues).filter(
          (item: string) => countStatusValues[item] > 1
        );
      }

      const newRows = rowsState.map(
        (row: PresetWorkingHoursDataType): PresetWorkingHoursDataType => {
          if (notUniqueStatus.length === 0) {
            row.label.isError = false;
          }

          const slicedRow = Object.keys(row).slice(1);
          for (const cell of slicedRow) {
            if (row[cell].value.length === 0) {
              row[cell].isError = true;
              row[cell].errorText = emptyValueError;
            }
          }

          // !unique status
          if (notUniqueStatus.length !== 0) {
            for (const cell of notUniqueStatus) {
              if (row.label.value.trim().toLowerCase() === cell) {
                row.label.isError = true;
                row.label.errorText = uniqueLabelError;
              }
            }
          }

          const time = row.time.value;
          if (!/^[\d]{1}[.][\d]{1,2}$/.test(time) && time.length) {
            row.time.isError = true;
            row.time.errorText = hoursPatternError;
          }
          if (/^[\d]{1,5}$/.test(time)) {
            row.time.isError = false;
            row.time.errorText = '';
          }
          if (
            /^[\d]{1}[.][\d]{1,2}$/.test(time) &&
            !Number.isInteger((Number(time) * 100) / 25)
          ) {
            row.time.isError = true;
            row.time.errorText = hoursValueError;
          }

          return row;
        }
      );

      setRowsState(newRows);

      let errCount = 0;
      newRows.forEach((row): void => {
        const slicedRow = Object.keys(row).slice(1);
        for (const cell of slicedRow) {
          if (row[cell].isError) {
            errCount += 1;
          }
        }
      });

      if (errCount === 0) {
        let sum = 0;
        newRows.forEach(({ time }): void => {
          sum += Number(time.value) * 100;
        });

        if (sum / 100 < HOURS_PER_DAY) {
          setIsLess(true);
          setIsEmptyTable(false);
          setIsModalOpen(true);
        } else if (sum / 100 > HOURS_PER_DAY) {
          setIsLess(false);
          setIsEmptyTable(false);
          setIsModalOpen(true);
        } else setIsValid(true);
      }

      if (isEmpty(newRows)) {
        setIsEmptyTable(true);
        setIsModalOpen(true);
      }
    };

    const handleModalCloseClick = (): void => {
      setIsModalOpen(false);
    };

    const handleModalSaveClick = (): void => {
      setIsValid(true);
      setIsModalOpen(false);
    };

    useEffect(() => {
      if (isValid) {
        const rowStateLabels: string[] = [];

        rowsState.forEach((user: PresetWorkingHoursDataType): void => {
          const label: string = user.label.value;
          const time: number = Number(user.time.value) * 3600;
          const rowsApiLabels: string[] = [];

          rowStateLabels.push(label);

          for (const key of rowsApi) {
            rowsApiLabels.push(key.label.value);
            if (
              key.label.value === label &&
              key.time.value !== user.time.value
            ) {
              dispatch(putPresetWorkingHours({ label, time }));
            }
          }

          if (!rowsApiLabels.includes(label)) {
            dispatch(postPresetWorkingHours({ label, time }));
          }
        });

        for (const key of rowsApi) {
          const label = key.label.value;
          if (!rowStateLabels.includes(label)) {
            dispatch(deletePresetWorkingHours(label));
          }
        }

        setIsEdit(false);
        setIsValid(false);
      }
    }, [isValid]);

    return (
      <div className='settings-table'>
        <div className='settings-table__wrapper'>
          <TableContainer component={Paper} className={editClassName}>
            <Table aria-label='table'>
              <TableHead>
                <TableRow>
                  {isEdit && <TableCell />}
                  {HEADER_CELLS.map((headCell) => (
                    <TableCell key={headCell}>{headCell}</TableCell>
                  ))}
                </TableRow>
              </TableHead>
              <TableBody>
                {rows.map(({ id, label, time }) => (
                  <TableRow key={id}>
                    {isEdit && (
                      <TableCell>
                        <Checkbox
                          className='edit__checkbox'
                          inputProps={{ 'aria-label': 'checkbox' }}
                          color='error'
                          onClick={() =>
                            handleCheckboxClick(id, selected, setSelected)
                          }
                        />
                      </TableCell>
                    )}
                    <TableCell>
                      {isEdit && (
                        <>
                          <FormControl fullWidth>
                            <Select
                              className={label.isError ? 'error-input' : ''}
                              native
                              defaultValue={label.value}
                              id='grouped-native-select'
                              name='label'
                              onChange={(e) =>
                                onChangeInput(e, id, rowsState, setRowsState)
                              }
                            >
                              <option value={optionValue1}>
                                {optionValue1}
                              </option>
                              <option value={optionValue2}>
                                {optionValue2}
                              </option>
                              <option value={optionValue3}>
                                {optionValue3}
                              </option>
                            </Select>
                          </FormControl>
                          {label.isError && (
                            <ErrorTooltip title={label.errorText} />
                          )}
                        </>
                      )}
                      {!isEdit && label.value}
                    </TableCell>
                    <TableCell>
                      {isEdit && (
                        <>
                          <input
                            className={time.isError ? 'error-input' : ''}
                            type='text'
                            value={time.value}
                            autoComplete='off'
                            name='time'
                            onChange={(e) =>
                              onChangeInput(e, id, rowsState, setRowsState)
                            }
                          />
                          {time.isError && (
                            <ErrorTooltip title={time.errorText} />
                          )}
                        </>
                      )}
                      {!isEdit && time.value}
                    </TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
          {isEditable && (
            <SettingsTableBtnsBlock
              isEdit={isEdit}
              isValid={isValid}
              setIsEdit={setIsEdit}
              addNewRow={addNewRow}
              setSelected={setSelected}
              rowsState={rowsState}
              selected={selected}
              setRowsState={setRowsState}
              rowsApi={copyRowsApi}
              checkValidation={checkValidation}
            />
          )}
        </div>
        {isModalOpen && (
          <PresetWorkingHoursModal
            onClose={handleModalCloseClick}
            modalNotification={modalNotification}
            onSave={handleModalSaveClick}
            isEmptyTable={isEmptyTable}
          />
        )}
      </div>
    );
  }
);

export default UsersContainer;

UsersContainer.displayName = 'UsersContainer';
