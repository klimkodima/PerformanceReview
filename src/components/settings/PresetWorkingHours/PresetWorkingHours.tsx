import { ReactElement } from 'react';

import PresetWorkingHoursContainer from './PresetWorkingHoursContainer';

const PresetWorkingHours = (): ReactElement => {
  return <PresetWorkingHoursContainer isEditable={false} />;
};

export default PresetWorkingHours;
