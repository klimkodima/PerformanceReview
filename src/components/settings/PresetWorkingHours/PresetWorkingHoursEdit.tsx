import { ReactElement } from 'react';

import PresetWorkingHoursContainer from './PresetWorkingHoursContainer';

const PresetWorkingHoursEdit = (): ReactElement => {
  return <PresetWorkingHoursContainer isEditable={true} />;
};

export default PresetWorkingHoursEdit;
