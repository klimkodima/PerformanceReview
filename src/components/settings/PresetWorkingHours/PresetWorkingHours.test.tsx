import { I18nextProvider } from 'react-i18next';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';

import store from 'src/store';
import i18n from 'src/i18n';
import { setPresetWorkingHours } from 'src/store/PresetWorkingHours';
import mockPresetWorkingHours from 'src/store/PresetWorkingHours/mockPresetWorkingHours.json';
import PresetWorkingHours from './PresetWorkingHours';
import PresetWorkingHoursEdit from './PresetWorkingHoursEdit';
import PresetWorkingHoursContainer from './PresetWorkingHoursContainer';

beforeEach(() => {
  store.dispatch(setPresetWorkingHours(mockPresetWorkingHours));
});

describe('PresetWorkingHours table tests', () => {
  test('PresetWorkingHours table snapshot', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <PresetWorkingHoursContainer isEditable={false} />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('PresetWorkingHours table has the elements', () => {
    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHours />
        </I18nextProvider>
      </Provider>
    );

    expect(getByRole('table')).toBeInTheDocument();
    expect(getAllByRole('columnheader')).toHaveLength(2);
    expect(getAllByRole('row')).toHaveLength(3);
  });

  test('PresetWorkingHours table Edit has the elements', () => {
    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    expect(getByRole('table')).toBeInTheDocument();
    expect(getAllByRole('columnheader')).toHaveLength(2);
    expect(getAllByRole('row')).toHaveLength(3);
    expect(getAllByRole('button')).toHaveLength(1);
  });

  test('PresetWorkingHours table enter and exit from edit mode', () => {
    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    userEvent.click(getByRole('button', { name: 'Edit Table' }));
    expect(getAllByRole('button')).toHaveLength(5);
    userEvent.click(getByRole('button', { name: 'Cancel' }));
    expect(getAllByRole('button')).toHaveLength(1);
  });

  test('Exit from edit mode if the PresetWorkingHours table is valid', () => {
    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    userEvent.click(getByRole('button', { name: 'Edit Table' }));
    const hoursField = getAllByRole('textbox');
    userEvent.clear(hoursField[0]);
    userEvent.type(hoursField[0], '4');
    userEvent.clear(hoursField[1]);
    userEvent.type(hoursField[1], '4');
    userEvent.click(getByRole('button', { name: 'Save' }));
    expect(getAllByRole('button')).toHaveLength(1);
  });

  test('PresetWorkingHours table - validation tests', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole, getAllByTestId, getByText } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    userEvent.click(getByRole('button', { name: 'Edit Table' }));
    const hoursField = getAllByRole('textbox');
    userEvent.clear(hoursField[0]);
    userEvent.type(hoursField[0], 'Error');

    const saveBtn = getByRole('button', { name: 'Save' });
    userEvent.click(saveBtn);
    expect(getAllByTestId('PriorityHighIcon')).toHaveLength(1);

    userEvent.clear(hoursField[0]);
    userEvent.type(hoursField[0], '0.15');
    userEvent.click(saveBtn);
    expect(getAllByTestId('PriorityHighIcon')).toHaveLength(1);

    userEvent.clear(hoursField[0]);
    userEvent.type(hoursField[0], '4');

    const addBtn = getByRole('button', { name: 'Add' });
    userEvent.click(addBtn);
    userEvent.click(saveBtn);
    expect(getAllByRole('row')).toHaveLength(4);
    expect(getAllByTestId('PriorityHighIcon')).toHaveLength(3);

    const checkboxes = getAllByRole('checkbox');
    expect(checkboxes).toHaveLength(3);
    userEvent.click(checkboxes[0]);
    userEvent.click(checkboxes[1]);
    userEvent.click(checkboxes[2]);
    const deleteBtn = getByRole('button', { name: 'Delete' });
    expect(deleteBtn).toHaveProperty('disabled', false);
    userEvent.click(deleteBtn);
    userEvent.click(saveBtn);

    expect(getByText(/The table cannot be empty!/i)).toBeInTheDocument();
    const cancelBtns = getAllByRole('button', { name: 'Cancel' });
    expect(cancelBtns).toHaveLength(2);
    userEvent.click(cancelBtns[1]);
    expect(getAllByRole('button', { name: 'Cancel' })).toHaveLength(1);
  });

  test('PresetWorkingHours table - put data', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole, getByText, queryByText } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    userEvent.click(getByRole('button', { name: 'Edit Table' }));
    const textField = getAllByRole('textbox');
    userEvent.clear(textField[0]);
    userEvent.type(textField[0], '10');
    userEvent.click(getByRole('button', { name: 'Save' }));
    expect(
      getByText(
        /The total duration exceeds 8 hours. Are you sure you want to save the table?/i
      )
    ).toBeInTheDocument();
    userEvent.click(getByRole('button', { name: 'Yes' }));
    expect(
      queryByText(
        /The total duration exceeds 8 hours. Are you sure you want to save the table?/i
      )
    ).toBeNull();
  });

  test('PresetWorkingHours table - delete data', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'portal');
    document.body.appendChild(root);

    const { getByRole, getAllByRole } = render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <PresetWorkingHoursEdit />
        </I18nextProvider>
      </Provider>
    );

    expect(getAllByRole('row')).toHaveLength(3);
    userEvent.click(getByRole('button', { name: 'Edit Table' }));
    const textField = getAllByRole('textbox');
    userEvent.clear(textField[0]);
    userEvent.type(textField[0], '10');
    const checkboxes = getAllByRole('checkbox');
    userEvent.click(checkboxes[0]);
    userEvent.click(getByRole('button', { name: 'Delete' }));
    expect(getAllByRole('row')).toHaveLength(2);
    userEvent.click(getByRole('button', { name: 'Save' }));
    userEvent.click(getByRole('button', { name: 'Yes' }));
  });
});
