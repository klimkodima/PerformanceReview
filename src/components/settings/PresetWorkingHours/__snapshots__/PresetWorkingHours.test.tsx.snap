// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PresetWorkingHours table tests PresetWorkingHours table snapshot 1`] = `
<div
  className="settings-table"
>
  <div
    className="settings-table__wrapper"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="table"
        className="MuiTable-root css-rqglhn-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Label
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Hours
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Audits
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              0.25
            </td>
          </tr>
          <tr
            className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              Support
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            >
              0.5
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
